src/views/open-api-docs-view/utils/get-version-switcher-props.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// Types
+import type { VersionSwitcherProps } from 'components/version-switcher'
+import type { OpenApiDocsVersionData } from '../types'
+
+/**
+ * Given version data and other OpenAPI docs details,
+ * Return version switcher dropdown props for use in `OpenApiDocsView`.
+ *
+ * Note: If there is only one version, we return null.
+ */
+export function getVersionSwitcherProps({
```

Comment:
This function does the server-side work of transforming the fetched version data (as well as some basic OpenAPI spec info for things like the project name) into a set of `VersionSwitcherProps`.

Note this work builds on versioned API docs implemented for HCP Packer. Namely, [fetch-cloud-api-version-data](https://github.com/hashicorp/dev-portal/blob/main/src/lib/api-docs/fetch-cloud-api-version-data/index.ts). The fetching of this version data happens at the _page_ level (this allows us to take different approaches to fetching versioned OpenAPI docs data if we want, as this is expected to differ between HCP APIs and other API docs!).